feature/bgp/gracefulrestart/otg_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go

```diff
@@ -0,0 +1,708 @@
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+	ctx := context.Background()
+
+	// Configure interface on the DUT
+	t.Log("Start DUT interface Config")
+	configureDUT(t, dut)
+	configureRoutePolicy(t, dut, "ALLOW", oc.RoutingPolicy_PolicyResultType_ACCEPT_ROUTE)
+
+	// Configure BGP+Neighbors on the DUT
+	t.Log("Configure BGP with Graceful Restart option under Global Bgp")
+	dutConfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	nbrList := buildNbrList(ateAS)
+	dutConf := bgpWithNbr(dutAS, nbrList, dut)
+	gnmi.Replace(t, dut, dutConfPath.Config(), dutConf)
+	fptest.LogQuery(t, "DUT BGP Config", dutConfPath.Config(), gnmi.Get(t, dut, dutConfPath.Config()))
+
+	// ATE Configuration.
+	t.Log("Start ATE Config")
+	config := configureATE(t, ate)
+
+	// Verify Port Status
+	t.Run("verifyDUTPorts", func(t *testing.T) {
+		t.Log("Verifying port status")
+		verifyPortsUp(t, dut.Device)
+	})
+	t.Run("VerifyBGPParameters", func(t *testing.T) {
+		t.Log("Check BGP parameters")
+		checkBgpStatus(t, dut)
+	})
+
+	// Starting traffic before graceful restart from DUT
+	t.Log("Starting traffic before Graceful restart trigger from DUT")
+	ate.OTG().StartTraffic(t)
+	startTime := time.Now()
+	t.Log("Send Traffic while GR timer counting down. Traffic should pass as BGP GR is enabled!")
+
+	// Send GNOI Graceful Restart Trigger from DUT to ATE
+	// Find the PID of BGP Daemon.
+	var pId uint64
+	pName := BGPDaemons[dut.Vendor()]
+
+	t.Run("FindBGPDaemonPid", func(t *testing.T) {
+		pId = findProcessByName(ctx, t, dut, pName)
+		if pId == 0 {
+			t.Fatalf("Couldn't find pid of gRIBI daemon '%s'", pName)
+		} else {
+			t.Logf("Pid of gRIBI daemon '%s' is '%d'", pName, pId)
+		}
+	})
+
+	// Kill gRIBI daemon through gNOI Kill Request.
+	t.Run("ExecuteGnoiKill", func(t *testing.T) {
+		// TODO - pid type is uint64 in oc-system model, but uint32 in gNOI Kill Request proto.
+		// Until the models are brought in line, typecasting the uint64 to uint32.
+		gNOIKillProcess(ctx, t, dut, pName, uint32(pId))
+		// Wait for a bit for gRIBI daemon on the DUT to restart.
+	})
+
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(ateDst.IPv4)
+
+	t.Run("VerifyBGPNOTEstablished", func(t *testing.T) {
+		t.Log("Waiting for BGP neighbor to go to ACTIVE state after GR Trigger to DUT...")
+		bgpState := oc.Bgp_Neighbor_SessionState_ACTIVE
+		_, ok := gnmi.Watch(t, dut, nbrPath.SessionState().State(), 2*time.Minute, func(val *ygnmi.Value[oc.E_Bgp_Neighbor_SessionState]) bool {
+			currState, ok := val.Val()
+			return ok && currState == bgpState
+		}).Await(t)
+		if !ok {
+			fptest.LogQuery(t, "BGP reported state", nbrPath.State(), gnmi.Get(t, dut, nbrPath.State()))
+			t.Errorf("BGP session did not go Down as expected")
+		}
+	})
```

Comment:
Should there be a mechanism here preventing the DUT/ATE to re-establish the BGP connection after killing the bgp daemon? It seems like this was included in the [new readme](https://github.com/openconfig/featureprofiles/commit/08c902105b58a58748a665646452976e566ae02a#diff-2ab3ef14592d0bf4292c57211f4ccf8d95041656e417fc2bab51d9f72bad53d9R34-R42). @jasdeep-hundal, thoughts?